pygmt/gridops.py

```diff
@@ -113,3 +113,115 @@ def grdcut(grid, **kwargs):
+    will not be used and the output can be within one half-width of the input edges.
+    If the filter is low-pass, then the output may be less frequently sampled than the input.
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+        outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    filter : str
+        Name of filter type you which to apply, followed by the width
+        b: Box Car; c: Cosine Arch; g: Gaussian; o: Operator; m: Median;
+        p: Maximum Likelihood probability; h: histogram
+        Example: F='m600' for a median filter with width of 600
+    distance : str
+        Distance flag, that tells how grid (x,y) rrlated to the filter width as follows:
+        flag = p: grid (px,py) with width an odd number of pixels; Cartesian distances.
+
+        flag = 0: grid (x,y) same units as width, Cartesian distances.
+
+        flag = 1: grid (x,y) in degrees, width in kilometers, Cartesian distances.
+
+        flag = 2: grid (x,y) in degrees, width in km, dx scaled by cos(middle y), Cartesian distances.
+
+        The above options are fastest because they allow weight matrix to be computed only once.
+        The next three options are slower because they recompute weights for each latitude.
+
+        flag = 3: grid (x,y) in degrees, width in km, dx scaled by cosine(y), Cartesian distance calculation.
+
+        flag = 4: grid (x,y) in degrees, width in km, Spherical distance calculation.
+
+        flag = 5: grid (x,y) in Mercator -Jm1 img units, width in km, Spherical distance calculation.
+
+    increment : str
+        x_inc [and optionally y_inc] is the grid spacing.
+        (http://docs.generic-mapping-tools.org/latest/grdfilter.html#i)
+    nans : Str or Number
+        Determine how NaN-values in the input grid affects the filtered output.
+        Values are i|p|r (http://docs.generic-mapping-tools.org/latest/grdfilter.html#n)
+    {R}
+    toggle : Bool
+        Toggle the node registration for the output grid so as to become the opposite of the input grid
+        (http://docs.generic-mapping-tools.org/latest/grdfilter.html#t)
+    {V}
+    {f}
+
+    Returns
+    -------
+    ret: xarray.DataArray or None
+        Return type depends on whether the *outgrid* parameter is set:
+        - xarray.DataArray if *outgrid* is not set
+        - None if *outgrid* is set (grid output will be stored in *outgrid*)
+
+    Usage
+    -------
+    pygmt.grdfilter('input.nc',F='m1600',D='4', G='filtered_output.nc')
+    Applies a filter of 1600km (full width) in the input.nc and returns a a filtered field (saved as netcdf)
+
+    smooth_field=pygmt.grdfiler(dataarray,F='g600',D='4')
+    Applies a gaussian smoothing filter of 600 km in the input data array,
+    and returns a filtered data array withthe smoothed field.
+    """
```

Comment:
```suggestion
    Examples
    --------
    >>> import os
    >>> import pygmt

    >>> # Apply a filter of 600km (full width) to the @earth_relief_30m file
    >>> # and return a filtered field (saved as netcdf)
    >>> pygmt.grdfilter(
    ...     grid="@earth_relief_30m",
    ...     filter="m600",
    ...     distance="4",
    ...     region=[150, 250, 10, 40],
    ...     spacing=0.5,
    ...     outgrid="filtered_pacific.nc"
    ... )
    >>> os.remove("filtered_pacific.nc")  # cleanup file

    >>> # Apply a gaussian smoothing filter of 600 km in the input data array,
    >>> # and returns a filtered data array with the smoothed field.
    >>> grid = pygmt.datasets.load_earth_relief()
    >>> smooth_field = pygmt.grdfilter(
    ...     grid=grid, filter="g600", distance="4"
    ... )

    """
```

Somewhat adapted from https://docs.generic-mapping-tools.org/6.1/grdfilter.html#examples, let me know if this looks ok.

Reply:
This looks good!